echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+    return weight_strata_aged_unaged , weight_strata
 
-def sum_strata_length_age_sex_weight( haul_data: pd.DataFrame ,
-                                      specimen_data: pd.DataFrame ,
-                                      length_intervals: np.ndarray ,
-                                      age_intervals: np.ndarray ):
-    
-    ### Sum weights from each station and strata 
-    weight_strata_station , weight_strata = sum_strata_weight( haul_data ,
-                                                               specimen_data )
+def compute_aged_unaged_proportions( specimen_data: pd.DataFrame ,
+                                     weight_strata: pd.DataFrame ):
+    """
+   Compute the overall weight proportions for aged and uanged fish within each stratum 
+
+    Parameters
+    ----------
+    specimen_data: pd.DataFrame
+        Dataframe containing specimen weights
+    weight_strata: pd.DataFrame
+        Dataframe contained summed weights of both aged and unaged fish
+    """   
+    ### Calculate adult:all age ratio for each stratum
+    # ---- Drop unaged fish
+    specimen_data_filtered = specimen_data[ specimen_data.sex != 'unsexed' ].dropna( how = 'any' , subset = 'age' ) 
+
+    # ---- Sum the weight of adult fish
+    specimen_adult = specimen_data_filtered[ specimen_data_filtered.age > 1 ].groupby( [ 'stratum_num' ] )[ 'weight' ].sum( )
+
+    # ---- Sum the weight of all fish
+    specimen_all = specimen_data_filtered.groupby( [ 'stratum_num' ] )[ 'weight' ].sum( ).to_frame( )
+
+    # ---- Calculate the ratio between the two groups
+    specimen_all[ 'weight_ratio' ] = ( specimen_adult / specimen_all.weight ).fillna( 0 )
+
+    ### Calculate the aged proportions of all fish
+    # ---- Merge with `weight_strata`
+    aged_proportions = specimen_all.reset_index( ).merge( weight_strata , on = [ 'stratum_num' ] )
+
+    # ---- Calculate the aged proportion of all fish
+    aged_proportions[ 'proportion_aged_weight_all' ] = (
+        aged_proportions.weight / aged_proportions.weight_stratum_all
+    )
+
+    # ---- Calculate the aged proportion of adult fish
+    aged_proportions[ 'proportion_aged_weight_adult' ] = (
+        aged_proportions.proportion_aged_weight_all * aged_proportions.weight_ratio
+    )
+
+    ### Calculate the unaged weight proportions
+    # ---- All fish
+    aged_proportions[ 'proportion_unaged_weight_all' ] = (
+        1.0 - aged_proportions.proportion_aged_weight_all
+    )
+
+    # ---- Adult fish
+    aged_proportions[ 'proportion_unaged_weight_adult' ] = (
+        1.0 - aged_proportions.proportion_aged_weight_adult
+    )
+
+    return aged_proportions.drop( [ 'weight_ratio' ] , axis = 1 )
+
+def compute_aged_weight_proportions( specimen_data: pd.DataFrame ,
+                                     length_intervals: np.ndarray ,
+                                     age_intervals: np.ndarray ):
+    """
+    Calculate length-age binned weight proportions    
```

Comment:
Thought it may be a good idea to include the data source
```suggestion
    Calculate length-age binned weight proportions based on specimen_df
```